src/view/App.re

```diff
@@ -29,13 +30,8 @@ let make = (~data: Store.state(Store.data),
     showHelp: false
   },
 
-  didMount: ({ send, state }) => {
-    ReasonReact.Update({
-      ...state,
-      worker: ref(Some(Worker.make(
-        ~onMessage  = {message => send(WorkerMessage(message))}
-      )))
-    })
+  didMount: ({ send }) => {
+    send(Init(message => send(WorkerMessage(message))))
```

Comment:
Is it necessary to pass the callback in here? I don't see what this closes over that isn't also available in the reducer?

Reply:
The `send` function? Afaik in the reducer you only have access to `action` and `state`... but maybe I'm missing something?

Reply:
Right, you'd have to use `SideEffects` and do another round through the reducer to update it, so better the way it is now.

Integrating RR components with external services really leaves something to be desired... Either that or I'm thinking about this all wrong...